doc/CONFIGURATION.md

```diff
@@ -42,6 +42,8 @@ See a [sample](./etc/pgmoneta.conf) configuration for running `pgmoneta` on `loc
 | tls_cert_file | | String | No | Certificate file for TLS. This file must be owned by either the user running pgmoneta or root. |
 | tls_key_file | | String | No | Private key file for TLS. This file must be owned by either the user running pgmoneta or root. Additionally permissions must be at least `0640` when owned by root or `0600` otherwise. |
 | tls_ca_file | | String | No | Certificate Authority (CA) file for TLS. This file must be owned by either the user running pgmoneta or root.  |
+| aes_key_length | 256 | int | No | The key length of AES (128, 192, 256) |
+| aes_mode | cbc | String | No | The operation mode of AES (cbc, ctr) |
```

Comment:
Likely easier for the user if it was just one configuration parameter

```
encryption = ...
```

with values

```
none (default)
aes (aes-256-cbc)
aes-256 (aes-256-cbc)
aes-192 (aes-192-cbc)
aes-128 (aes-128-cbc)
aes-256-cbc
aes-192-cbc
aes-128-cbc
aes-256-ctr
aes-192-ctr
aes-128-ctr
```

The values can be split internally

Reply:
I will merge the configuration parameters into one. 

I have a question about the **none** option. In current codebase (without my modification), the ``pgmoneta_encrypt()`` is called from ``update_user()`` and ``add_user()``, and the ``pgmoneta_decrypt()`` is called from ``pgmoneta_read_users_configuration()`` and ``pgmoneta_read_admins_configuration()``. I read those codes and it seems the aes-256-cbc is always on, there's no switch to on/off the encryption.

If I am right, what would you suggest where and when to read the configuration and setup the server logic? I currently put the read config and determine encryption algorithm codes in ``get_cipher()`` and call it from ``aes_encrypt()`` and ``aes_decrypt()``. The problem is  ``get_cipher()`` is executed everytime the ``pgmoneta_encrypt()`` or ``pgmoneta_decrypt()`` is called. I think it is a kind of waste or we may change the configuration while the service is running?

Reply:
Sounds good.

The existing functionality will continue to use AES_256_CBC - so maybe add the encryption type to the `aes_encrypt(int aes_type,  / aes_decrypt(int aes_type, ` functions. 

Once that is done you will need to start to encrypt / decrypt the main files for the backup - I would start with the WAL files and add a `pgmoneta_encrypt_wal(d)` call here https://github.com/pgmoneta/pgmoneta/blob/main/src/main.c#L1426. If `config->encryption` is `ENCRYPTION_NONE` then the function returns at once. Otherwise files are encrypted - that aren't encrypted yet. Only `pgmoneta_encrypt|decrypt_wal|data` will use `get_cipher` to pass the correct constant.

Hope this helps !

Reply:
Thanks for the feedback.
So we're gonna have 2 sets of keys, one for the admin and one for the WAL. The encryption config is only for the WAL and the admin will always use aes-256-cbc. Am I correct?

Reply:
The existing functionality will always use AES_256_CBC so we don't need a configuration option for that

Reply:
If you are asking about the master key - it will be used for both areas

Reply:
Okay, I think I know what to do, thank you!